src/api/utils/summaries.py

```diff
@@ -7,40 +9,46 @@
 )
 
 
-def update_project_summaries(project_id, *args, **kwargs):
-    update_project_summary_site(project_id)
-    update_project_summary_sample_event(project_id)
+def update_project_summaries(project_id, skip_test_project=True, *args, **kwargs):
+    update_project_summary_site(project_id, skip_test_project)
+    update_project_summary_sample_event(project_id, skip_test_project)
 
 
 def update_project_summary_site(project_id, skip_test_project=True):
     if (
         skip_test_project
         and Project.objects.filter(pk=project_id, status=Project.TEST).exists()
     ):
+        SummarySiteModel.objects.filter(project_id=project_id).delete()
```

Comment:
Why not just do this once, before doing anything else, and then not repeat it below?

Reply:
delete has to happen after the sql table query, which doesn't happen in this spot